src/smallstack.rs

```diff
@@ -92,33 +92,31 @@ impl<'a> VM<'a> {
         &["carry", "invert"] => carry = !carry,
 
         &["call", "primitive", "say"] => if let Some(Expr::Str(arg)) = stack.pop() {
-          println!("hey {}", arg);
+          println!("{}", arg);
         } else {
           panic!("VM error: incorrect arguments to `say`");
         },
 
+        &["call", "primitive", "concat"] => if let
+            (Some(Expr::Str(arg1)), Some(Expr::Str(arg2)))
+            =
+            (stack.pop(), stack.pop()) {
+          stack.push(Expr::Str(arg1 + &arg2));
```

Comment:
Totally agree
On Thu, 18 Aug 2016 at 10:44, ven <notifications@github.com> wrote:

> In src/smallstack.rs
> <https://github.com/vendethiel/smallstack/pull/12#discussion_r75278082>:
>
> >          } else {
> >            panic!("VM error: incorrect arguments to `say`");
> >          },
> >
> > +        &["call", "primitive", "concat"] => if let
> > +            (Some(Expr::Str(arg1)), Some(Expr::Str(arg2)))
> > +            =
> > +            (stack.pop(), stack.pop()) {
> > +          stack.push(Expr::Str(arg1 + &arg2));
>
> somewhat feeling uneasy about this.
>
> i guess we should have &["call", "primitive", x] => and then try to find
> the function in some map (the stdlib).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vendethiel/smallstack/pull/12/files/02047ac90b74d517afdd13f68a2d4c0405904d42#r75278082>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAhBdbsTOVCdHCj-M_VKUrdRt64C9djjks5qhCmJgaJpZM4Jm5R2>
> .
>